src/System.Device.Gpio/System/Device/Gpio/Drivers/Libgpiod/V1/LibGpiodV1Driver.cs

```diff
@@ -5,22 +5,24 @@
 using System.Threading;
 using System.Runtime.InteropServices;
 using System.Collections.Concurrent;
+using System.Device.Gpio.Drivers;
+using System.Device.Gpio.Interop.Unix.libgpiod.V1;
 using System.Diagnostics;
-using System.Net.NetworkInformation;
+using LibgpiodV1 = Interop.LibgpiodV1;
 
-namespace System.Device.Gpio.Drivers;
+namespace System.Device.Gpio.System.Device.Gpio.Drivers.Libgpiod.V1;
 
 /// <summary>
-/// This driver uses the Libgpiod library to get user-level access to the gpio ports.
+/// This driver uses libgpiod v0/1 to get user-level access to the gpio ports.
 /// It superseeds the SysFsDriver, but requires that libgpiod is installed. To do so, run
 /// "sudo apt install -y libgpiod-dev".
 /// </summary>
-public class LibGpiodDriver : UnixDriver
+public class LibGpiodV1Driver : UnixDriver
```

Comment:
This rename is a major breaking change, and the class name is used directly by clients, so I think this shouldn't be renamed, even if it makes the names a bit missleading.

Reply:
Yeah, as @pgrawehr points out, unfortunately we cannot take this breaking change as it would have high impact (a lot of folks are creating GpioControllers today like: `... = new GpioController(new LibGpiodDriver())`). I so much regret that we didn't make LibGpiodDriver a Factory pattern as that would have been ideal, and we would just have V1 and V2 internal concrete implementations that would be returned by a Create method. Anyway, that ship has sailed now, and in terms of design, I don't think callers would ever want to select whether they use V1 or V2 of libgpiod driver, so no point on making changes in exposing those classes. From the consumer point of view, they should only select that they want LibGpiodDriver, but then for that to internally figure out if it should use V1 or V2 depending on what native library is installed.

For that reason, our public API should probably just keep LibGpiodDriver as the one entry point class, and for that to redirect internally to V2 stuff if needed.

Reply:
I think we should be careful with adding new automatic detection here, this has mostly caused issues (remember the mess we created with the RaspberryPi3Driver?) We should try to move the detection to separate classes going forward, e.g. to the `Board` namespace.

Reply:
I can't say much about that. Except that it is not impossible that distros will provide libgpiodv2 by default at some point.
By then at the latest there should be automatic detection or the possibility for users to determine the version.

Reply:
I have pushed changes that contain the solution of keeping LibGpiodDriver and it's namepsace to not break things. In the constructor it will resolve the proper driver version and act as a bridge to it. How the automatic detection is implemented is not yet final obviously and is discussed in another comment

Reply:
I think we can agree on the current implementation that does not contain breaking changes. If you have comments on automatic driver version loading, please make another comment, I will resolve this conversation